_posts/2019-09-11-how-to-stay-up-to-date.markdown

```diff
@@ -0,0 +1,138 @@
+
+### Start treating deprecations warnings as errors
+
+In Rails you can configure this option, so everytime a deprecation warning occurs you can address it as soon as you can.
+This will help you to fix the issue immediately and be prepared for the next version jump.
+
+You will need to add the following configuration in your `config/environments/test.rb` and `config/environments/test.rb` files.
+
+```ruby
+  # Raise error on deprecation¬
+  config.active_support.deprecation = :raise
+```
+
+### Dual Boot
+
+One of the first things we do at FastRuby.io when we start a new Rails upgrade project is to setup a "Dual Boot" Gemfile.
+What this setup will do is allow you to have two Gemfile.lock files, one for the latest version of Rails and one for your current version of Rails.
+
+With this setup you will be able to configure your CI to run both versions of Rails. As long as you "mark" the Rails master build job as "optional" it won't block your PR merges. You will have the chance to see how your test behaves with the latest version of Rails.
+
+We even wrote a [blog post](https://www.fastruby.io/blog/upgrade-rails/dual-boot/dual-boot-with-rails-6-0-beta.html) about this setup, you should check it out.
+
+### Check for outdated gems
+
+You can use bundle outdated to check outdated gems in your current bundle. While `bundle outdated` is quite useful it doesn't give us details relative to our own `Gemfile`. In Ombu Labs we like to use  [next_rails](https://rubygems.org/gems/next_rails/versions/1.0.0) because this gem will give us a set of tools to check for outdated gems, check incompatibilities, and even bootstrap an upgrade project.
+
+### Automated updates
+[Dependabot](https://dependabot.com/) is a bot that will open a PR in your Github repository whenever a dependency is outdated. Dependabot was acquired by Github and it's now free of charge.
+You can still install Dependabot from the Github Marketplace until it's integrated into GitHub
+
+<div style="text-align: center; width: 500px;">
+  <img src="/blog/assets/images/dependabot/dependabot.png">
+</div>
+
+We know not all of you use Github but if it happens that you use it, dependabot would be a really nice adition to your tools to keep you up to date.
+
+### Bundler audit
+
+Your application has dependencies. Those dependencies can have security vulnerabilities(CVE's) and malicious persons can take advantage of this.
+
+A tool that we use to make sure our clients dependencies are free of gems with CVE's is [bundler-audit](https://github.com/rubysec/bundler-audit), bundler-audit will scan your current bundle and will compare it with an internal database of know CVE's. If bundler-leak finds any gem affected by a CVE it will dislplay all the information of the affected gem.
```

Comment:
```suggestion
A tool that we use to make sure our clients dependencies are free of gems with CVE's is [bundler-audit](https://github.com/rubysec/bundler-audit), bundler-audit will scan your current bundle and will compare it with an internal database of know CVE's. If bundler-audit finds any gem affected by a CVE it will display all the information of the affected gem.
```